src/cli/common.rs

```diff
@@ -406,6 +406,45 @@ pub(super) fn list_items(
     Ok(utils::ExitCode(0))
 }
 
+pub(super) fn list_items_custom(
```

Comment:
I'm afraid this is still too repetitive. Can `list_items()` take a list of components instead so that you can reuse that part of the logic?